contracts/modules/Checkpoint/EtherDividendCheckpoint.sol

```diff
@@ -121,11 +121,16 @@ contract EtherDividendCheckpoint is DividendCheckpoint {
             _name
           )
         );
-        for (uint256 i = 0; i < _excluded.length; i++) {
-            excludedSupply = excludedSupply.add(ISecurityToken(securityToken).balanceOfAt(_excluded[i], _checkpointId));            
-            dividends[dividends.length - 1].dividendExcluded[_excluded[i]] = true;
+
+        for (uint256 j = 0; j < _excluded.length; j++) {
+            require (_excluded[j] != address(0), "Invalid address");
+            for (uint256 i = j + 1; i < _excluded.length; i++) {

```

Comment:
Instead of the two loops, could we just have:
`require(!dividends[dividendIndex].dividendExcluded[_excluded[i]], "duped address")`
?

Reply:
(same for ERC20 & EtherDividends)

Reply:
Reading will storage will cost ~250 gas so this will become effective only is the list size is greater than ~20.
I'll still make the change because it lowers the code complexity.
Thanks for pointing it out.